### PR TITLE
Introduce clang-format

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

**Comment by ethteck** [comment]
@nanaian have you given this more thought? I still think the benefits of having auto-formatting don't outweigh the cons atm, without lots of engineering work to make it fit nicely with EVTs. And considering I think where we last ended that convo was that we would put EVTs in separate files with their own DSL, that seems like it blocks this anyway

**Comment by nanaian** [comment]
As @marijnvdwerf said perhaps it would be beneficial to have maintainers look at the diff clang-format spits out as a suggestion but not fail the PR for it. That is, we'd use it to find and suggest fixes to nits. Does that make sense?

**Comment by ethteck** [comment]
Kinda, but I find the output kinda blah, like "look at these lines"

**Comment by nanaian** [comment]
Mm, that's true. Perhaps we close this PR for now but keep the branch and revisit once we have a better solution for evts

**Comment by marijnvdwerf** [comment]
> Kinda, but I find the output kinda blah, like "look at these lines"

That's why I added the `clang-format` command to the action. It's output does show the actual diff:
image

**Comment by ethteck** [comment]
i sheep

**Comment by ethteck** [comment]
I don't think we want this whole PR - just the clang format file by itself

**Comment by nanaian** [comment]
> I don't think we want this whole PR - just the clang format file by itself

I've moved the job to the `lint` workflow which can continue to be disabled until we solve the EVT issue

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    Priority:        10
+  # everything else
+  - Regex:           '.*'
+    Priority:        1
+    SortPriority:    0
+
+AllowShortFunctionsOnASingleLine: None
+---
+
+...
```
Comment (marijnvdwerf) [inline .clang-format:61]: What are these dots?

Comment (ethteck) [reply .clang-format:61]: image

**Pull request merged**